File: Assets/Scripts/GameManager.cs
using UnityEngine;
using System.Collections;

public class GameManager : MonoBehaviour {
	
	public GameObject characterPrefab;
	
	private ILevel GameLevel;
	
	// Use this for initialization
	void Start () {
		GameLevel = new Level(new SnakeMap(), new StoneHallRenderer(), characterPrefab);
	}
	
	void OnGUI () {
    	if (GUI.Button (new Rect (10,10,150,100), "Generate")) {
			GameLevel.GenerateNewLevel();
		}
	}	

}

File: Assets/Scripts/GameManager.js
#pragma strict

public class GameManager extends MonoBehaviour {

	public var characterPrefab: GameObject;
	
	private var GameLevel: ILevel;
	
	// Use this for initialization
	function Start () {
		GameLevel = new Level(new SnakeMap(), new StoneHallRenderer(), characterPrefab);
	}
	
	function OnGUI () {
    	if (GUI.Button (new Rect (10,10,75,25), "Generate")) {
			GameLevel.GenerateNewLevel();
		}		
	}

}